### PR TITLE
make it possible to disable labeling the namespaces

### PR DESCRIPTION
**Pull request by sparkoo**
Signed-off-by: Michal Vala <mvala@redhat.com>

### What does this PR do?
Adding che property to disable che-server to try to label the workspace namespaces. If it is enabled, we do not fail, but log warn message.

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18301

### Documentation PR
https://github.com/eclipse/che-docs/pull/1694

### How to test this PR?

1. deploy with `che.infra.kubernetes.namespace.label=true`
1. you should see either warning that che-server can't label the namespace, or if che SA has the permissions, the namespace should be labeled
1. set the `che.infra.kubernetes.namespace.label=false`
1. you should no longer see the warning, and new namespaces should not be labeled

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by sparkoo** [comment]
[crw-ci-test]

**Comment by sparkoo** [comment]
[ci-build]

**Comment by sparkoo** [comment]
[crw-ci-test]

**Review by skabashnyuk — APPROVED** [review]

**Review by ibuziuk — APPROVED** [review]
I believe for Hosted Che we will need to set `che.infra.kubernetes.namespace.label` to false

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java [hunk 1]
```diff
@@ -187,7 +190,12 @@ protected void label(Namespace namespace, Map<String, String> ensureLabels)
                   .build());
     } catch (KubernetesClientException kce) {
       if (kce.getCode() == 403) {
-        LOG.debug("Can't label the namespace due to lack of permissions ¯\\_(ツ)_/¯");
+        LOG.warn(
+            "Can't label the namespace due to lack of permissions. Grant to `che` ServiceAccount "
```
Comment (metlos) [inline infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java]: Grammar nazi suggests:
> Can't label the namespace due to lack of permissions. Grant cluster-wide permissions to \`get\` and \`update\` the \`namespaces\` to the \`che\` service account (Che operator might have already prepared a cluster role called \`che-namespace-editor\` for this, depending on its configuration). Alternatively, consider disabling the feature by setting \`che.infra.kubernetes.namepsace.label\` to \`false\`.

**Review by metlos — APPROVED** [review]
LGTM, maybe some minor wording suggestions...

File: infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java (hunk 1 above)
Comment (sparkoo) [reply infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesNamespace.java]: always welcome 👍🏼

**Pull request merged**